java/ql/lib/qlpack.yml

```diff
@@ -1,7 +1,8 @@
 name: codeql/java-all
-version: 0.0.2
+version: 0.0.3
+groups: java
+library: true
 dbscheme: config/semmlecode.dbscheme
 extractor: java
-library: true
 dependencies:
-    codeql/java-upgrades: 0.0.2
+    codeql/java-upgrades: "*"
```

Comment:
So, the move from an explicit version to a `*` is ok because we are OK with users who try to compile `codeql/java-all` outside of this workspace getting an error?

Reply:
Wait, let me think about this. For library packs this probably isn't right, because we want a real version dependency in the published library pack.

Reply:
Ah...yes. I think we need a hard version.

This complicates the upgrade process since we will need to bump the upgrades dependency as well.